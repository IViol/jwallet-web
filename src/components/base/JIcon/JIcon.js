// @flow

import React from 'react'
import classNames from 'classnames'
import { iconsUI } from 'utils/sprite'

export type JIconColor = 'white' | 'blue' | 'gray' | 'sky' | 'red' | 'black'

type Props = {
  name: string,
  color: ?JIconColor,
}

function JIcon({
  name, color,
}: Props) {
  const iconData = iconsUI[`${name}-usage`]
  const hasFill = name.indexOf('use-fill') !== -1

  return (
    <svg
      className={classNames(
        'j-icon',
        color && `-${color}`,
        hasFill && '-use-fill')}
      width={iconData.width}
      height={iconData.height}
    >
      <use xlinkHref={iconData.url} />
    </svg>
  )
}

JIcon.defaultProps = {
  color: null,
}

export default JIcon
