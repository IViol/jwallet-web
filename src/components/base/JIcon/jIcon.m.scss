@import 'styles/colors';

.core {
  fill: none;
  cursor: inherit;
  stroke-width: 1.5px;
  display: inline-block;
  transition: stroke 0.3s ease 0s;
}

.fill {
  stroke-width: 0;
}

.empty {
  width: 24px;
  height: 24px;
  background-color: $tomato;
}

.white {
  stroke: $white;

  &.fill {
    fill: $white;
  }
}

.blue {
  stroke: $blue;

  &.fill {
    fill: $blue;
  }
}

.gray {
  stroke: $dusk;

  &.fill {
    fill: $dusk;
  }
}

.sky {
  stroke: $bright-sky-blue;

  &.fill {
    fill: $bright-sky-blue;
  }
}

.red {
  stroke: $tomato;

  &.fill {
    fill: $tomato;
  }
}

.black {
  stroke: $main-black;

  &.fill {
    fill: $main-black;
  }
}
