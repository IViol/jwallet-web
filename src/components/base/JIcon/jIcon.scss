.j-icon {
  all: initial;

  @include hover-animation;

  display: flex;
  background-repeat: no-repeat;
  background-position: center;
  cursor: inherit;

  /**
   * Small icons
   */
  &.-small {
    width: $icon-small-size;
    height: $icon-small-size;
    background-size: $icon-small-size $icon-small-size;

    &.-arrow {
      &.-blue {
        background-image: $icon-small-arrow-blue;
      }

      &.-white {
        background-image: $icon-small-arrow-white;
      }
    }

    &.-arrow-down {
      transform: rotate(90deg);

      &.-blue {
        background-image: $icon-small-arrow-blue;
      }

      &.-sky {
        background-image: $icon-small-arrow-sky;
      }

      &.-white {
        background-image: $icon-small-arrow-white;
      }
    }

    &.-checkbox {
      background-image: $icon-small-checkbox-white;

      &.-blue {
        background-image: $icon-small-checkbox-blue;
      }
    }

    &.-close {
      background-image: $icon-small-close-white;

      &.gray {
        background-image: $icon-small-close-gray;
      }
    }

    &.-expand {
      &.blue {
        background-image: $icon-small-expand-blue;
      }

      &.-gray {
        background-image: $icon-small-expand-gray;
      }

      &.-white {
        background-image: $icon-small-expand-white;
      }
    }

    &.-eye {
      &.-gray {
        background-image: $icon-small-eye-gray;
      }
    }

    &.-info {
      background-image: $icon-small-info-white;

      &.-blue {
        background-image: $icon-small-info-blue;
      }

      &.-gray {
        background-image: $icon-small-info-gray;
      }
    }

    &.-info-clear {
      &.-gray {
        background-image: $icon-small-info-clear-gray;
      }
    }

    &.-lock {
      &.-gray {
        background-image: $icon-small-lock-gray;
      }
    }

    &.-menu-favorite {
      background-image: $icon-small-menu-favorite-white;
    }

    &.-menu-receive {
      background-image: $icon-small-menu-receive-white;
    }

    &.-menu-send {
      background-image: $icon-small-menu-send-white;
    }

    &.-narrow {
      transform: rotate(180deg);

      &.blue {
        background-image: $icon-small-expand-blue;
      }

      &.-gray {
        background-image: $icon-small-expand-gray;
      }

      &.-white {
        background-image: $icon-small-expand-white;
      }
    }

    &.-plus {
      &.-blue {
        background-image: $icon-small-plus-blue;
      }

      &.-sky {
        background-image: $icon-small-plus-sky;
      }

      &.-white {
        background-image: $icon-small-plus-white;
      }
    }

    &.-repeat {
      background-image: $icon-small-repeat-white;

      &.-blue {
        background-image: $icon-small-repeat-blue;
      }
    }

    &.-settings {
      background-image: $icon-small-settings-white;

      &.-blue {
        background-image: $icon-small-settings-blue;
      }

      &.-gray {
        background-image: $icon-small-settings-gray;
      }
    }

    &.-star {
      background-image: $icon-small-star-white;

      &.-blue {
        background-image: $icon-small-star-blue;
      }
    }
  }

  /**
   * Medium icons
   */
  &.-medium {
    width: $icon-medium-size;
    height: $icon-medium-size;
    background-size: $icon-medium-size $icon-medium-size;

    &.-arrow {
      background-image: $icon-medium-arrow-white;

      &.-blue {
        background-image: $icon-medium-arrow-blue;
      }
    }

    &.-arrow-header {
      background-image: $icon-medium-arrow-header-white;

      &.-blue {
        background-image: $icon-medium-arrow-header-blue;
      }

      &.-gray {
        background-image: $icon-medium-arrow-header-gray;
      }
    }

    &.-close-header {
      background-image: $icon-medium-close-header-white;

      &.-gray {
        background-image: $icon-medium-close-header-gray;
      }
    }

    &.-copy {
      &.-blue {
        background-image: $icon-medium-copy-blue;
      }

      &.-sky {
        background-image: $icon-medium-copy-sky;
      }
    }

    &.-dots {
      background-image: $icon-medium-dots-white;
    }

    &.-download {
      &.-blue {
        background-image: $icon-medium-download-blue;
      }

      &.-sky {
        background-image: $icon-medium-download-sky;
      }
    }

    &.-info {
      background-image: $icon-medium-info-white;

      &.-blue {
        background-image: $icon-medium-info-blue;
      }

      &.-gray {
        background-image: $icon-medium-info-gray;
      }
    }

    &.-key-import {
      background-image: $icon-medium-key-import-white;

      &.-blue {
        background-image: $icon-medium-key-import-blue;
      }

      &.-gray {
        background-image: $icon-medium-key-import-gray;
      }
    }

    &.-key-new {
      background-image: $icon-medium-key-new-white;

      &.-blue {
        background-image: $icon-medium-key-new-blue;
      }

      &.-gray {
        background-image: $icon-medium-key-new-gray;
      }
    }

    &.-link {
      background-image: $icon-medium-link-white;

      &.-blue {
        background-image: $icon-medium-link-blue;
      }
    }

    &.-lock {
      background-image: $icon-medium-lock-white;

      &.-blue {
        background-image: $icon-medium-lock-blue;
      }

      &.-gray {
        background-image: $icon-medium-lock-gray;
      }
    }

    &.-logout {
      background-image: $icon-medium-logout;
    }

    &.-menu-bg {
      background-image: $icon-medium-menu-bg-white;
    }

    &.-mnemonic {
      background-image: $icon-medium-mnemonic;
    }

    &.-mnemonic-read {
      background-image: $icon-medium-mnemonic-read;
    }

    &.-private-key {
      background-image: $icon-medium-private-key;
    }

    &.-private-key-read {
      background-image: $icon-medium-private-key-read;
    }

    &.-search {
      background-image: $icon-medium-search-white;

      &.-gray {
        background-image: $icon-medium-search-gray;
      }
    }
  }

  /**
   * Large icons
   */
  &.-large {
    width: $icon-large-size;
    height: $icon-large-size;
    background-size: $icon-large-size $icon-large-size;

    &.-transaction-receive {
      background-image: $icon-large-transaction-receive;
    }

    &.-transaction-receive-time {
      background-image: $icon-large-transaction-receive-time;
    }

    &.-transaction-send {
      background-image: $icon-large-transaction-send;
    }

    &.-transaction-send-time {
      background-image: $icon-large-transaction-send-time;
    }
  }
}
