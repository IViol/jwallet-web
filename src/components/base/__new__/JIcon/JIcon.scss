.JIcon {
  all: initial;
  display: block;
  background-repeat: no-repeat;
  background-position: left center;

  // Variants

  &.-transparent {
    opacity: 0.2;
  }

  // Sizes

  &.-small {
    width: 12px;
    height: 12px;
    background-size: 12px 12px;
  }

  &.-medium {
    width: 24px;
    height: 24px;
    background-size: 24px 24px;
  }

  &.-large {
    width: 36px;
    height: 36px;
    background-size: 36px 36px;
  }

  &.-extra-large {
    width: 48px;
    height: 48px;
    background-size: 48px 48px;
  }

  &.-huge {
    width: 200px;
    height: 200px;
    background-size: 200px 200px;
  }


  // SVG

  &.-plus {
    background-image: url("../../../../public/icons/small/plus-blue.svg");
  }

  &.-menu-send {
    background-image: url("../../../../public/icons/small/menu-send-white.svg");
  }

  &.-menu-receive {
    background-image: url("../../../../public/icons/small/menu-receive-white.svg");
  }

  &.-menu-favorite {
    background-image: url("../../../../public/icons/small/menu-favorite-white.svg");
  }

  &.-template {
    background-image: url("../../../../public/icons/small/star-blue.svg");
  }

  &.-repeat {
    background-image: url("../../../../public/icons/small/repeat-blue.svg");
  }

  &.-arrow-down {
    background-image: url("../../../../public/icons/small/arrow-blue.svg");
    transform: rotate(90deg);
  }

  &.-arrow-back {
    background-image: url("../../../../public/icons/small/arrow-white.svg");
  }

  &.-expand-gray {
    background-image: url("../../../../public/icons/small/expand-gray.svg");
  }

  &.-close {
    background-image: url("../../../../public/icons/small/close-gray.svg");
  }

  &.-menu-bg {
    background-image: url("../../../../public/icons/medium/menu-bg-white.svg");
  }

  &.-logout {
    background-image: url("../../../../public/icons/medium/logout.svg");
  }

  &.-search {
    background-image: url("../../../../public/icons/medium/search-gray.svg");
  }

  &.-close-popup {
    background-image: url("../../../../public/icons/medium/close-popup-gray.svg");
  }

  &.-login-back {
    background-image: url("../../../../public/icons/medium/arrow-popup-white.svg");
  }

  &.-key-new {
    background-image: url("../../../../public/icons/medium/key-new-white.svg");
  }

  &.-key-import {
    background-image: url("../../../../public/icons/medium/key-import-white.svg");
  }

  &.-private-key {
    background-image: url("../../../../public/icons/medium/private-key.svg");
  }

  &.-private-key-read {
    background-image: url("../../../../public/icons/medium/private-key-read.svg");
  }

  &.-mnemonic {
    background-image: url("../../../../public/icons/medium/mnemonic.svg");
  }

  &.-mnemonic-read {
    background-image: url("../../../../public/icons/medium/mnemonic-read.svg");
  }

  &.-dots-white {
    background-image: url("../../../../public/icons/medium/dots-white.svg");
  }

  &.-arrow-forward {
    background-image: url("../../../../public/icons/medium/arrow-white.svg");
  }

  &.-ant-token {
    background-image: url("../../../../public/icons/large/tokens/ant.svg");
  }

  &.-transaction-send {
    background-image: url("../../../../public/icons/extra-large/transaction-send.svg");
  }

  &.-transaction-send-time {
    background-image: url("../../../../public/icons/extra-large/transaction-send-time.svg");
  }

  &.-transaction-receive {
    background-image: url("../../../../public/icons/extra-large/transaction-receive.svg");
  }
  
  &.-transaction-receive-time {
    background-image: url("../../../../public/icons/extra-large/transaction-receive-time.svg");
  }

  &.-key {
    background-image: url("../../../../public/icons/huge/key.svg");
  }
}
