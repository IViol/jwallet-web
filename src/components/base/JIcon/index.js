// @flow strict

import JIcon from './JIcon'

export default JIcon
