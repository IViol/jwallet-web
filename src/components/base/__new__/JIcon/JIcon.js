/* @flow */

import React from 'react'
import { pure } from 'recompose'
import cx from 'classnames'
import { prop } from 'ramda'

import './JIcon.scss'

type Props = {
  name: string,
  size: 'small' | 'medium' | 'large' | 'extra-large' | 'huge',
}

const JIcon = ({ name, size }: Props) => (
  <div className={cx(
    'JIcon',
    prop(size, {
      huge: '-huge',
      large: '-large',
      small: '-small',
      medium: '-medium',
      'extra-large': '-extra-large',
    }),
    `-${name}`,
  )}
  />
)

export default pure(JIcon)
