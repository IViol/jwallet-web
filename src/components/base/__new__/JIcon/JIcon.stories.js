
/* @flow */

import React from 'react'
import { storiesOf } from '@storybook/react'

import JIcon from './JIcon'

storiesOf('JIcon', module)
  .add('Small close', () => (
    <JIcon
      size='small'
      name='close'
    />
  ))
  .add('Medium logout', () => (
    <JIcon
      size='medium'
      name='logout'
    />
  ))
  .add('Large Ant token', () => (
    <JIcon
      size='large'
      name='ant-token'
    />
  ))
  .add('Extra large received transaction', () => (
    <JIcon
      size='extra-large'
      name='transaction-receive'
    />
  ))
