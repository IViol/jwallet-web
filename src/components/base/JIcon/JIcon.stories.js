// @flow

import React from 'react'
import { storiesOf } from '@storybook/react'
import {
  withKnobs,
  select,
} from '@storybook/addon-knobs'

import { JIcon } from 'components/base'

// $FlowFixMe
const filesSmall = require.context(
  '../../../public/assets/icons/sprite-pack/small', true, /.*\.svg$/,
)

// $FlowFixMe
const filesMedium = require.context(
  '../../../public/assets/icons/sprite-pack/medium', true, /.*\.svg$/,
)

// $FlowFixMe
const filesLarge = require.context(
  '../../../public/assets/icons/sprite-pack/large', true, /.*\.svg$/,
)

// $FlowFixMe
const filesXLarge = require.context(
  '../../../public/assets/icons/sprite-pack/xlarge', true, /.*\.svg$/,
)

// $FlowFixMe
const filesColored = require.context(
  '../../../public/assets/icons/sprite-colored', true, /.*\.svg$/,
)

const iconsSmall = filesSmall
  .keys()
  .map(
    filepath =>
      filesSmall(filepath).default.id.replace(/-usage$/, ''),
  )
const iconsMedium = filesMedium
  .keys()
  .map(
    filepath =>
      filesMedium(filepath).default.id.replace(/-usage$/, ''),
  )
const iconsLarge = filesLarge
  .keys()
  .map(
    filepath =>
      filesLarge(filepath).default.id.replace(/-usage$/, ''),
  )
const iconsXLarge = filesXLarge
  .keys()
  .map(
    filepath =>
      filesXLarge(filepath).default.id.replace(/-usage$/, ''),
  )

const iconsColored = filesColored
  .keys()
  .map(
    filepath =>
      filesColored(filepath).default.id.replace(/-usage$/, ''),
  )

const AVAILABLE_COLORS = [
  null,
  'white',
  'blue',
  'gray',
  'sky',
  'red',
  'black',
]

// this array is created specifically to be sorted here
// eslint-disable-next-line fp/no-mutating-methods
const AVAILABLE_ICONS = [
  ...iconsSmall,
  ...iconsMedium,
  ...iconsLarge,
  ...iconsXLarge,
].sort()

storiesOf('JIcon', module).addDecorator(withKnobs)
  .add('Customizable', () => (
    <div>
      <h2 className='title'>Customizable example</h2>
      <div className='icon-demo-container'>
        <JIcon
          name={select('Name', AVAILABLE_ICONS, AVAILABLE_ICONS[0])}
          color={select('Color', AVAILABLE_COLORS, 'blue')}
          className={select(
            'Class name',
            {
              'none': null,
              'icon-with-magenta-color-and-margins': 'icon-with-magenta-color-and-margins',
            },
            null,
          )}
        />
      </div>
    </div>
  ))
  .add('List of all icons', () => {
    const allIconsListColorSelectValue = select('Color', AVAILABLE_COLORS, 'blue')

    return (
      <div>
        <section>
          <h2 className='title'>Small</h2>
          <ul className='icons-list'>
            {iconsSmall.map(name => (
              <li key={name}>
                <JIcon
                  name={name}
                  color={allIconsListColorSelectValue}
                />
                <span>{name}</span>
              </li>
            ))}
          </ul>
        </section>
        <section>
          <h2 className='title'>Medium</h2>
          <ul className='icons-list'>
            {iconsMedium.map(name => (
              <li key={name}>
                <JIcon
                  name={name}
                  color={allIconsListColorSelectValue}
                />
                <span>{name}</span>
              </li>
            ))}
          </ul>
        </section>
        <section>
          <h2 className='title'>Large</h2>
          <ul className='icons-list'>
            {iconsLarge.map(name => (
              <li key={name}>
                <JIcon
                  name={name}
                  color={allIconsListColorSelectValue}
                />
                <span>{name}</span>
              </li>
            ))}
          </ul>
        </section>
        <section>
          <h2 className='title'>X-Large</h2>
          <ul className='icons-list'>
            {iconsXLarge.map(name => (
              <li key={name}>
                <JIcon
                  name={name}
                  color={allIconsListColorSelectValue}
                />
                <span>{name}</span>
              </li>
            ))}
          </ul>
        </section>
        <section>
          <h2 className='title'>Colored</h2>
          <ul className='icons-list'>
            {iconsColored.map(name => (
              <li key={name}>
                <JIcon
                  name={name}
                />
                <span>{name}</span>
              </li>
            ))}
          </ul>
        </section>
      </div>
    )
  })
