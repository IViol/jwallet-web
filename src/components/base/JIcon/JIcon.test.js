/* @flow */

import React from 'react'
import renderer from 'react-test-renderer'

import JIcon from './JIcon'

describe('COMPONENT: JIcon', () => {
  test('Small close', () => {
    expect(renderer.create(
      <JIcon
        size='small'
        name='close'
      />
    ).toJSON()).toMatchSnapshot()
  })
})
