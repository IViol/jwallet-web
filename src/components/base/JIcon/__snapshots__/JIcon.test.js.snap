// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT: JIcon Small close 1`] = `
<div
  className="JIcon -small -close"
/>
`;
